src/functions-and-arrays.js

```diff
@@ -1,73 +1,462 @@
 // Iteration #1: Find the maximum
 
+const maxOfTwoNumbers = (num1, num2) => {
+  if (num1 > num2) {
+    return num1;
+  } else if (num1 < num2) {
+    return num2;
+  } else if (num1 === num2) {
+    return num1;
+  }
+};
+
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot",
+];
+
+function findLongestWord(arr) {
+  if (arr.length === 0) {
+    return null;
+  } else {
+    newArr = arr[0];
+    for (var i = 0; i < arr.length - 1; i++) {
+      if (newArr.length < arr[i + 1].length) {
+        newArr = arr[i + 1];
+      }
+    }
+    return newArr;
+  }
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+const sumNumbers = (arr) => {
+  let count = 0;
+  for (let i = 0; i < arr.length; i++) {
+    count += arr[i];
+  }
+  return count;
+};
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+const averageNumbers = (arr) => {
+  let count = 0;
+  for (let i = 0; i < arr.length; i++) {
+    count += arr[i];
+  }
+  return count / arr.length;
+};
+
 // Level 2: Array of strings
-const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+const wordsArr = [
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace",
+];
+
+const averageWordLength = (arr) => {
+  let sumArr = 0;
+  for (let i = 0; i < arr.length; i++) {
+    sumArr += arr[i].length;
+  }
+  return sumArr / arr.length;
+};
 
 // Iteration #5: Unique arrays
 const wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+  "crab",
+  "poison",
+  "contagious",
+  "simple",
+  "bring",
+  "sharp",
+  "playground",
+  "poison",
+  "communion",
+  "simple",
+  "bring",
 ];
 
+const uniquifyArray = (arr) => {
+  var newArr = {};
+  return arr.filter(function (item) {
+    return newArr.hasOwnProperty(item) ? false : (newArr[item] = true);
+  });
```

Comment:
Interesting approach. You could also do indexOf() method here, that way you don't even that newArr variable.